docs/dev/developer-guides/transactions/fee-model.md

```diff
@@ -39,3 +36,67 @@ So the tips to make the most out of the zkSync fee system are the following:
 - **Reuse as many storage slots as possible.** Only the state diff is published on Ethereum.
 - **Users should share as many storage slots as possible.** If 100 users update a storage slot of your contract in a single block, the diff will be published only once. In the future, we will introduce reimbursement for the users, so that the costs for updating shared storage slots are split between the users.
 - **Reuse the contract code if possible.** On Ethereum, avoiding constructor parameters and putting them into constants reduces some of the gas costs upon contract deployment. On zkSync the opposite is true: deploying the same bytecode for contracts, while changing only constructor parameters can lead to substantial fee savings.
+
+
+## Gas estimation during transaction: For custom and paymaster accounts
+
+On Ethereum, there is a constant of 21000 gas that should cover all the intrinsic costs of processing a transaction: checking the signature and updating the nonce of the account. 
+
+On zkSync Era, it's quite different, because we support custom accounts that may take a slightly different (typically a bit higher) number of gas than EOAs. 
+We provide a convenient way for anyone to estimate the cost of a transaction regardless of the type of account. 
+### Changes in the `validateTransaction`
+
+Right now, the `validateTransaction` fails if the signature is not correct and returns `success=true` otherwise, while the `returndata` is discarded completely. We will migrate to a different approach, where the `validateTransaction` method to be considered successful must return some magic string. Any other `returndata` will be considered invalid and will be rejected by the system.  
+
+Now, `validateTransaction` method of the AA (or the paymaster `validateAndPayForPaymasterTransaction` method), should always try to do the same amount of computation (and especially storage accesses) as it would have done if the transaction was indeed validated correctly. By default, the operator will provide a transaction structure that contains as much information as is available to the operator at the time of the fee estimation. As a substitute for the signature, an invalid 65-byte ECDSA signature will be used. The `DefaultAccount` (the one that is used by EOAs) will do as many operations as possible (including signature verification) and only return `bytes4(0)` instead of magic. If, for instance, the wallet is a custom account, with multiple signers, then it may want to simulate the validation of the signatures of all the provided signers.
+
+So roughly the code of the validation step of each account will look the following way:
```

Comment:
Please use this section as a guide for external developers on how they should build their custom accounts. The doc that I've sent to you was about how we built DefaultAccount (EOAs).

At the very least the we should rather use `should ` instead of `will`, etc. Generally we should use Present Simple here (`do/does`) or recommendations (`should`)